app/dao/fact_notification_status_dao.py

```diff
@@ -291,3 +291,81 @@ def fetch_stats_for_all_services_by_date_range(start_date, end_date, include_fro
+        Template.is_precompiled_letter.label('is_precompiled_letter'),
+        extract('month', FactNotificationStatus.bst_date).label('month'),
+        extract('year', FactNotificationStatus.bst_date).label('year'),
+        func.sum(FactNotificationStatus.notification_count).label('count')
+    ).join(
+        Template, FactNotificationStatus.template_id == Template.id
+    ).filter(
+        FactNotificationStatus.service_id == service_id,
+        FactNotificationStatus.bst_date >= start_date,
+        FactNotificationStatus.bst_date <= end_date,
+    ).group_by(
+        FactNotificationStatus.template_id,
+        Template.name,
+        Template.template_type,
+        Template.is_precompiled_letter,
+        extract('month', FactNotificationStatus.bst_date).label('month'),
+        extract('year', FactNotificationStatus.bst_date).label('year'),
+    )
+
+    if start_date <= datetime.utcnow() <= end_date:
+        today = get_london_midnight_in_utc(datetime.utcnow())
+        month = get_london_month_from_utc_column(Notification.created_at)
+
+        stats_for_today = db.session.query(
+            Notification.template_id.label('template_id'),
+            Template.name.label('name'),
+            Template.template_type.label('template_type'),
+            Template.is_precompiled_letter.label('is_precompiled_letter'),
+            extract('month', month).label('month'),
+            extract('year', month).label('year'),
+            func.count().label('count')
+        ).join(
+            Template, Notification.template_id == Template.id,
+        ).filter(
+            Notification.created_at >= today,
+            Notification.service_id == service_id,
+            # we don't want to include test keys
+            Notification.key_type != KEY_TYPE_TEST
+        ).group_by(
+            Notification.template_id,
+            Template.hidden,
+            Template.name,
+            Template.template_type,
+            month
+        )
+
+        all_stats_table = stats.union_all(stats_for_today).subquery()
+        query = db.session.query(
+            all_stats_table.c.template_id,
+            all_stats_table.c.name,
+            all_stats_table.c.is_precompiled_letter,
+            all_stats_table.c.template_type,
+            func.cast(all_stats_table.c.month, Integer).label('month'),
+            func.cast(all_stats_table.c.year, Integer).label('year'),
+            func.cast(func.sum(all_stats_table.c.count), Integer).label('count'),
+        ).group_by(
+            all_stats_table.c.template_id,
+            all_stats_table.c.name,
+            all_stats_table.c.is_precompiled_letter,
+            all_stats_table.c.template_type,
+            all_stats_table.c.month,
+            all_stats_table.c.year,
+        ).order_by(
+            all_stats_table.c.year,
+            all_stats_table.c.month,
+            all_stats_table.c.name
+        )
+    else:
+        query = stats
```

Comment:
if the range doesn't cover today, then the query won't be ordered. is this okay?

Reply:
No - good point, I'll change that, and add a test